### PR TITLE
Move `GetLclOffs` method to `GenTreeLclVarCommon`.

### PR DESCRIPTION
**Pull request by sandreenko**
Move `GetLclOffs` offset method to `GenTreeLclVarCommon` to simplify blocks that are asking working both with Fields and Vars.

If I understood correctly  it is what was proposed in https://github.com/dotnet/runtime/pull/39424#pullrequestreview-451007759

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @AndyAyersMS @CarolEidt @dotnet/jit-contrib 

This is a refactoring PR for https://github.com/dotnet/runtime/pull/39424#pullrequestreview-451007759

File: src/coreclr/src/jit/gentree.cpp [hunk 1]
```diff
@@ -19350,6 +19346,18 @@ regNumber GenTree::ExtractTempReg(regMaskTP mask /* = (regMaskTP)-1 */)
     return genRegNumFromMask(tempRegMask);
 }
 
+uint16_t GenTreeLclVarCommon::GetLclOffs() const
```
Comment (AndyAyersMS) [inline src/coreclr/src/jit/gentree.cpp:19349]: Header comment?

File: src/coreclr/src/jit/codegenarm64.cpp [hunk 2]
```diff
@@ -4793,7 +4790,7 @@ void CodeGen::genStoreLclTypeSIMD12(GenTree* treeNode)
     regNumber operandReg = genConsumeReg(op1);
 
     // Need an addtional integer register to extract upper 4 bytes from data.
-    regNumber tmpReg = treeNode->GetSingleTempReg();
+    regNumber tmpReg = lclVar->GetSingleTempReg();
```
Comment (AndyAyersMS) [inline src/coreclr/src/jit/codegenarm64.cpp:4793]: Can you fix the typo in the line above? addtional -> additional

**Review by AndyAyersMS — APPROVED** [review]
Looks good, just some commenting nits.

File: src/coreclr/src/jit/codegenarm64.cpp (hunk 2 above)
Comment (sandreenko) [reply src/coreclr/src/jit/codegenarm64.cpp:4793]: Done.

File: src/coreclr/src/jit/gentree.cpp (hunk 1 above)
Comment (sandreenko) [reply src/coreclr/src/jit/gentree.cpp:19356]: Added.

**Pull request merged**